modelcontextprotocol/tools/assets.py

```diff
@@ -60,24 +62,71 @@ def update_assets(
 
         # Create assets with updated values
         assets = []
+        # Maintain separate lists for readme updates to avoid confusion with other asset attribute updates.
+        # readme_update_assets: Readme assets to update.
+        # readme_update_parent_assets: Assets that were updated with readme.
+        readme_update_assets = []
+        readme_update_parent_assets = []
         index = 0
         for updatable_asset in updatable_assets:
             type_name = updatable_asset.type_name
+            qualified_name = updatable_asset.qualified_name
             asset_cls = getattr(
                 __import__("pyatlan.model.assets", fromlist=[type_name]), type_name
             )
             asset = asset_cls.updater(
                 qualified_name=updatable_asset.qualified_name,
                 name=updatable_asset.name,
             )
-            setattr(asset, attribute_name.value, attribute_values[index])
-            assets.append(asset)
+
+            # Special handling for README updates
+            if attribute_name == UpdatableAttribute.README:
+                response_readme_fetch = (
+                    FluentSearch()
+                    .select()
+                    .where(CompoundQuery.asset_type(asset_cls))
+                    .where(asset_cls.QUALIFIED_NAME.eq(qualified_name))
+                    .include_on_results(asset_cls.README)
+                    .include_on_relations(Readme.DESCRIPTION)
+                    .execute(client=client)
+                )
+
+                if first := response_readme_fetch.current_page():
```

Comment:
Can you check the `:=` operator ? Is is native to python?

Reply:
This is a native python operator known as `walrus` opeartor introduced in [Python 3.8](https://www.geeksforgeeks.org/walrus-operator-in-python-3-8/)
This will not cause issue

The code runs locally without any runtime errors